src/files.c

```diff
@@ -82,27 +79,49 @@ bool is_marked_incompleted(char *file_contents, int file_contents_size) {
         char c2 = file_contents[i + 1];
         char c3 = file_contents[i + 2];
 
-        if (c1 == x1 && c2 == x2 && c3 == x3) return true;
+        if (c1 == x1 && c2 == x2 && c3 == x3)
+            return true;
     }
 
     return false;
 }
 
 void add_dir(FileCollection **dirs, File *files, int file_ct, int dir_idx) {
+
+    // Free dir if not null
+    if (dirs[dir_idx] != NULL) {
+        if (dirs[dir_idx]->files != NULL) {
+            for (size_t e = 0; e < dirs[dir_idx]->file_ct; e++) {
+                free(dirs[dir_idx]->files[e].file_path);
```

Comment:
I do not like this. It would be nice to move the allocation to the beginning. Then we could free it just at the end.